sdk/core/Azure.Core/src/Shared/GZipUtf8JsonRequestContent.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.IO;
+using System.IO.Compression;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal class GZipUtf8JsonRequestContent: RequestContent
+    {
+        private GZipStream _gzip;
+        private readonly MemoryStream _stream;
+        private readonly RequestContent _content;
+
+        public Utf8JsonWriter JsonWriter { get; }
+
+        public GZipUtf8JsonRequestContent(RequestContent content)
+        {
+            _stream = new MemoryStream();
+            _gzip = new GZipStream(_stream, CompressionMode.Compress, true);
+            _content = Create(_stream);
+            JsonWriter = new Utf8JsonWriter(_gzip);
+            content.WriteTo(_gzip, default);
```

Comment:
This may not be needed by clients, but it allows for scenarios where the payload has already been built for some reason without being GZipped. We can probably remove for v1.